mlos_bench/mlos_bench/optimizers/mlos_core_optimizer.py

```diff
@@ -221,18 +222,20 @@ def register(
             # TODO: Specify (in the config) which metrics to pass to the optimizer.
             # Issue: https://github.com/microsoft/MLOS/issues/745
             self._opt.register(
-                configs=df_config,
-                scores=pd.DataFrame([registered_score], dtype=float),
+                observations=Observations(
+                    config=df_config,
+                    score=pd.DataFrame([registered_score], dtype=float),
+                )
             )
         return registered_score
 
     def get_best_observation(
         self,
     ) -> Union[Tuple[Dict[str, float], TunableGroups], Tuple[None, None]]:
-        (df_config, df_score, _df_context) = self._opt.get_best_observations()
-        if len(df_config) == 0:
+        best_observations = self._opt.get_best_observations()
+        if len(best_observations.config) == 0:
```

Comment:
Since for `Observations` `len(config) == len(scores)`, let's add `__len__` to `Observations` so this can be `len(best_observations) == 0`